lib/Dialect/TritonGPU/Transforms/Pipeliner/MatmulLoopPipeline.cpp

```diff
@@ -96,8 +96,9 @@ createAsyncCopy(scf::ForOp &forOp, tt::LoadOp loadOp, Value alloc,
     mask = convertBlockLayout(src);
   }
 
-  SmallVector<Value> copyOffsets = {insertIdx, zero, zero};
   tt::MemDescType allocTy = alloc.getType().cast<tt::MemDescType>();
+  SmallVector<Value> copyOffsets(allocTy.getRank(), zero);
+  copyOffsets[0] = insertIdx;
```

Comment:
nit: to be honest I find the previous version of code easier to read

Reply:
The old version was wrong, though.

Is there a different way you'd like me to write it?

Reply:
ah because the rank may not be 3?

Reply:
oh I see, ok yeah that was an overfit to matmul pipeline that didn't get fixed